include/flatbuffers/flatbuffers.h

```diff
@@ -1265,7 +1271,7 @@ class Verifier FLATBUFFERS_FINAL_CLASS {
     // Call T::Verify, which must be in the generated code for this type.
     return Verify<uoffset_t>(buf_) &&
       reinterpret_cast<const T *>(buf_ + ReadScalar<uoffset_t>(buf_))->
-        Verify(*this);
+        Verify(*this) && GetSize();
```

Comment:
If we don't test for entire buffer you'll get contradictorios results:
VerifyMonsterBuffer will return true but Verifier::GetSize returns 0 ... which is not ok, that said, IMHO VerifyMonsterBuffer should return false in this case.